### PR TITLE
Escape percents

### PR DESCRIPTION
**Pull request by damccorm**
Right now, we unescape a bunch of variables to allow users to encode reserved values in ## commands. This means, however, that currently there's not a way to represent values like `%3B` - this allows for a complete encoding by also encoding `%25`.

Note that the task-lib already does this - https://github.com/microsoft/azure-pipelines-task-lib/blob/0d9223f07484efb6168c67d3632bfa6526c401c7/node/taskcommand.ts#L106

This change comes with a little risk. If anyone is expecting `%25` to return `%25` instead of `%` they'll be broken, that's why I put this behind a knob. We should eventually turn this on by default, I mention targeting January 2021 in the doc.

### CODE REVIEW

**Comment by damccorm** [comment]
This is failing CI and my local build because of some myget.org issues unrelated to the change. It shouldn't go in until we're able to build obviously, but I'd appreciate some feedback on the principle behind this change

**Comment by alex-peck** [comment]
Can we add some tests?

File: src/Microsoft.VisualStudio.Services.Agent/Command.cs [hunk 1]
```diff
@@ -12,11 +13,10 @@ public sealed class Command
         private const string LoggingCommandPrefix = "##vso[";
         private static readonly EscapeMapping[] s_escapeMappings = new[]
         {
-            // TODO: What about %?
```
[the remark this answers is not in the document]
Comment (damccorm) [reply src/Microsoft.VisualStudio.Services.Agent/Command.cs:15]: I'm not sure

**Comment by damccorm** [comment]
> Can we add some tests?

Yeah - I still can't build, but I will when our dependencies start working 😢

**Comment by damccorm** [comment]
After discussing with @mjroghelia decided to leave this off at first with the intent of turning it on down the line if it doesn't seem like it will break too many people.

**Review by mjroghelia — APPROVED** [review]

**Pull request merged**